lib/SIL/DynamicCasts.cpp

```diff
@@ -862,8 +862,7 @@ namespace {
       OptionalTypeKind sourceOptKind;
```

Comment:
Since you're not using sourceOptKind anywhere, you can use the version of getAnyOptionalObjectType() that takes no arguments

Reply:
I'll fix this in a follow-up.